### PR TITLE
MGMT-13141: assisted-installer-controller shouldn't try to update BMH if the cluster platform is not BareMetal

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-13141](https://issues.redhat.com//browse/MGMT-13141): assisted-installer-controller shouldn't try to update BMH if the cluster platform is not BareMetal

### CODE REVIEW

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 1]
```diff
@@ -667,29 +667,46 @@ func (c controller) UpdateBMHs(ctx context.Context, wg *sync.WaitGroup) {
 		c.log.Infof("Finished UpdateBMHs")
 		wg.Done()
 	}()
+
+	var cluster *models.Cluster
+	var err error
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller.go:672]: We get the cluster in the `SetReadyState` you can get it from there and just not start this goroutine.

Comment (tsorya) [reply src/assisted_installer_controller/assisted_installer_controller.go:672]: Optional yes

File: src/main/assisted-installer-controller/assisted_installer_main.go [hunk 2]
```diff
@@ -144,8 +144,12 @@ func main() {
 	wg.Add(1)
 	go assistedController.PostInstallConfigs(mainContext, &wg)
 	wg.Add(1)
-	go assistedController.UpdateBMHs(mainContext, &wg)
-	wg.Add(1)
+
+	if *cluster.Platform.Type != models.PlatformTypeBaremetal {
+		logger.Infof("Cluster platform is not %s, skipping BMH", models.PlatformTypeBaremetal)
+		go assistedController.UpdateBMHs(mainContext, &wg)
```
Comment (omertuc) [inline src/main/assisted-installer-controller/assisted_installer_main.go:150]: shouldn't this be outside the `if`?

Comment (tsorya) [reply src/main/assisted-installer-controller/assisted_installer_main.go:150]: right :(

Comment (tsorya) [reply src/main/assisted-installer-controller/assisted_installer_main.go:149]: fixed

**Pull request merged**